compiler-core/src/language_server/engine.rs

```diff
@@ -267,6 +270,107 @@ where
         })
     }
 
+    pub fn document_symbol(
+        &mut self,
+        params: lsp::DocumentSymbolParams,
+    ) -> Response<Vec<DocumentSymbol>> {
+        self.respond(|this| {
+            let mut symbols = vec![];
+            let Some(module) = this.module_for_uri(&params.text_document.uri) else {
+                return Ok(symbols);
+            };
+            let line_numbers = LineNumbers::new(&module.code);
+
+            for definition in &module.ast.definitions {
+                match definition {
+                    #[allow(deprecated)]
```

Comment:
We can not specify the `deprecated` field to avoid needing to allow deprecated fields?

Reply:
I think the problem is that not specifying the field translates to specifying it as `None` in the Rust side, whereas `Some(...)` would be (wrongly, since it is deprecated) specifying it. But their usage of `#[deprecated(...)]` doesn't distinguish between both cases, since it applies to any usage of the field...

Is there an alternative you're thinking of?

Reply:
Isn't the deprecated field the one called `deprecated`, which we do set to None?

If it's some other field could we put the attribute on the field itself to avoid making this entire clause? With a comment explaining the situation too

Reply:
The attribute being in the entire clause was a mistake, thanks for pointing that out. The clause used to just return a `DocumentSymbol` directly, which is why the attribute was added above it, but I added more stuff afterwards and forgot to move the attribute. I've fixed that now.

Unfortunately, it seems that adding the attribute to the field itself doesn't work, so I kept it above the `DocumentSymbol` constructor. I added a comment on the first one explaining the situation.